### PR TITLE
Enforce SSH Git URLs in install flux & gitops apply commands

### LINKED ISSUE #1212: `gitops apply` git urls should be ssh
**What happened?**

Tried to run `gitops apply` with http urls as parameters and it fails. 

**What you expected to happen?**

`eksctl` should enforce the right format of the git parameters and return a meaningful error if the format is not met.

**How to reproduce it?**

Run `eksctl gitops apply` with git urls in http format, rather than ssh format.

### PR DESCRIPTION
**Pull request by marccarre**
### Description

Fixes #1212, specifically, implements `2. (d)` mentioned [here](https://github.com/weaveworks/eksctl/issues/1212#issuecomment-526163555) by @2opremio.
Some clean-up and factorisation where possible/easy/relevant in the process.

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] ~Added/modified documentation as required (such as the `README.md`, and `examples` directory)~
- [x] Manually tested

### Manual tests

```console
$ EKSCTL_EXPERIMENTAL=true ./eksctl install flux
[✖]  please supply a valid --git-url argument: empty Git URL

$ EKSCTL_EXPERIMENTAL=true ./eksctl install flux --git-url meh
[✖]  please supply a valid --git-url argument: invalid Git URL

$ EKSCTL_EXPERIMENTAL=true ./eksctl install flux --git-url https://github.com/marccarre/my-gitops-repo.git
[✖]  please supply a valid --git-url argument: got a HTTP(S) Git URL, but eksctl currently only supports SSH Git URLs

$ EKSCTL_EXPERIMENTAL=true ./eksctl install flux --git-url git@github.com:marccarre/my-gitops-repo.git
[... No error on the Git URL...]

$ EKSCTL_EXPERIMENTAL=true ./eksctl install flux --help
Bootstrap Flux, installing it in the cluster and initializing its manifests in the specified Git repository

Usage: eksctl install flux [flags]

Flux installation flags:
      --git-url string                    SSH URL of the Git repository to be used by Flux, e.g. git@github.com:<github_org>/<repo_name>
[...]
```

```console
$ EKSCTL_EXPERIMENTAL=true ./eksctl gitops apply --cluster gitops --quickstart-profile app-dev --git-url meh
[✖]  please supply a valid --git-url argument: invalid Git URL

$ EKSCTL_EXPERIMENTAL=true ./eksctl gitops apply --cluster gitops --quickstart-profile app-dev --git-url https://github.com/marccarre/my-gitops-repo.git
[✖]  please supply a valid --git-url argument: got a HTTP(S) Git URL, but eksctl currently only supports SSH Git URLs

$ EKSCTL_EXPERIMENTAL=true ./eksctl gitops apply --cluster gitops --quickstart-profile app-dev --git-url git@github.com:marccarre/my-gitops-repo.git
[... No error on the Git URL...]

$ EKSCTL_EXPERIMENTAL=true ./eksctl gitops apply --help
Setting up GitOps and apply a Quick Start profile

Usage: eksctl gitops apply [flags]

General flags:
      --quickstart-profile string         name or URL of the Quick Start profile. For example, app-dev.
      --git-url string                    SSH URL of the Git repository that will contain the cluster components, e.g. git@github.com:<github_org>/<repo_name>
[...]
```

### CODE REVIEW

File: pkg/ctl/gitops/apply.go [hunk 1]
```diff
@@ -73,8 +73,8 @@ func doApplyGitops(cmd *cmdutils.Cmd, opts options) error {
 		return errors.New("please supply a valid gitops Quick Start URL or name in --quickstart-profile")
 	}
 
-	if opts.gitOptions.URL == "" {
-		return errors.New("please supply a valid --git-url argument")
+	if err := opts.gitOptions.ValidateURL(); err != nil {
+		return errors.Wrapf(err, "please supply a valid --git-url argument")
```
Comment (cPu1) [inline pkg/ctl/gitops/apply.go]: Nit:
```suggestion
		return errors.Wrap(err, "please supply a valid --git-url argument")
```

File: pkg/ctl/install/flux.go [hunk 2]
```diff
@@ -23,10 +23,10 @@ func installFluxCmd(cmd *cmdutils.Cmd) {
 	)
 	var opts flux.InstallOpts
 	cmd.SetRunFuncWithNameArg(func() error {
-		if opts.GitURL == "" {
-			return errors.New("please supply a valid --git-url argument")
+		if err := opts.GitOptions.ValidateURL(); err != nil {
+			return errors.Wrapf(err, "please supply a valid --git-url argument")
```
Comment (cPu1) [inline pkg/ctl/install/flux.go]:
```suggestion
			return errors.Wrap(err, "please supply a valid --git-url argument")
```

File: pkg/ctl/install/flux.go [hunk 3]
```diff
@@ -68,17 +68,17 @@ func installFluxCmd(cmd *cmdutils.Cmd) {
 	})
 
 	cmd.FlagSetGroup.InFlagSet("Flux installation", func(fs *pflag.FlagSet) {
-		fs.StringVar(&opts.GitURL, "git-url", "",
-			"URL of the Git repository to be used by Flux, e.g. git@github.com:<github_org>/flux-get-started")
-		fs.StringVar(&opts.GitBranch, "git-branch", "master",
+		fs.StringVar(&opts.GitOptions.URL, "git-url", "",
+			"SSH URL of the Git repository to be used by Flux, e.g.: git@github.com:<github_org>/<repo_name>")
```
Comment (cPu1) [inline pkg/ctl/install/flux.go]: Nit: I haven't seen a colon (`:`) used after `e.g.` before, I think we can remove it
```suggestion
			"SSH URL of the Git repository to be used by Flux, e.g. git@github.com:<github_org>/<repo_name>")
```

File: pkg/git/git.go [hunk 4]
```diff
@@ -46,6 +46,25 @@ type Options struct {
...
+		return errors.New("invalid Git URL")
+	}
+	if !o.isSSHURL() {
+		return errors.New("got a HTTP(S) Git URL, but eksctl currently only supports SSH Git URLs")
+	}
+	return nil
+}
+
+func (o Options) isSSHURL() bool {
+	return strings.HasPrefix(o.URL, "git@") || strings.HasPrefix(o.URL, "ssh://")
```
Comment (cPu1) [inline pkg/git/git.go]: For more accurate parsing, we should use the same library call `IsGitURL` uses: `giturls.Parse`
and check for the `ssh` scheme.

**Comment by marccarre** [comment]
### Full end-to-end test

From this branch, having:
```diff
$ git diff
diff --git a/examples/01-simple-cluster.yaml b/examples/01-simple-cluster.yaml
index 18389273..36f96d1a 100644
--- a/examples/01-simple-cluster.yaml
+++ b/examples/01-simple-cluster.yaml
@@ -4,13 +4,19 @@ apiVersion: eksctl.io/v1alpha5
 kind: ClusterConfig
 
 metadata:
-  name: cluster-1
-  region: eu-north-1
+  name: marcgitops
+  region: ap-northeast-1
 
 nodeGroups:
   - name: ng-1
     instanceType: m5.large
-    desiredCapacity: 1
+    minSize: 1
+    maxSize: 2
+    iam:
+      withAddonPolicies:
+        albIngress: true
+        autoScaler: true
+        cloudWatch: true
 
 cloudWatch:
     clusterLogging:
```

```console
$ EKSCTL_EXPERIMENTAL=true ./eksctl -f examples/01-simple-cluster.yaml gitops apply --quickstart-profile app-dev --git-url git@github.com:marccarre/my-gitops-repo.git --cluster marcgitops
[ℹ]  Generating public key infrastructure for the Helm Operator and Tiller
[ℹ]    this may take up to a minute, please be patient
[!]  Public key infrastructure files were written into directory "/var/folders/24/d3mml6bn20nftpt91cfldq1h0000gn/T/eksctl-helm-pki070380737"
[!]  please move the files into a safe place or delete them
[ℹ]  Generating manifests
[ℹ]  Cloning git@github.com:marccarre/my-gitops-repo.git
Cloning into '/var/folders/24/d3mml6bn20nftpt91cfldq1h0000gn/T/eksctl-install-flux-clone-982193452'...
remote: Enumerating objects: 65, done.
remote: Counting objects: 100% (65/65), done.
remote: Compressing objects: 100% (50/50), done.
remote: Total 207 (delta 15), reused 64 (delta 15), pack-reused 142
Receiving objects: 100% (207/207), 90.18 KiB | 263.00 KiB/s, done.
Resolving deltas: 100% (55/55), done.
[ℹ]  Writing Flux manifests
[ℹ]  created "Namespace/flux"
[ℹ]  Applying Helm TLS Secret(s)
[ℹ]  created "flux:Secret/flux-helm-tls-cert"
[ℹ]  created "flux:Secret/tiller-secret"
[!]  Note: certificate secrets aren't added to the Git repository for security reasons
[ℹ]  Applying manifests
[ℹ]  created "CustomResourceDefinition.apiextensions.k8s.io/helmreleases.helm.fluxcd.io"
[ℹ]  created "flux:ConfigMap/flux-helm-tls-ca-config"
[ℹ]  created "flux:Deployment.extensions/tiller-deploy"
[ℹ]  created "flux:Service/tiller-deploy"
[ℹ]  created "flux:Secret/flux-git-deploy"
[ℹ]  created "flux:Deployment.apps/flux"
[ℹ]  created "flux:ServiceAccount/tiller"
[ℹ]  created "ClusterRoleBinding.rbac.authorization.k8s.io/tiller"
[ℹ]  created "flux:ServiceAccount/helm"
[ℹ]  created "flux:Role.rbac.authorization.k8s.io/tiller-user"
[ℹ]  created "kube-system:RoleBinding.rbac.authorization.k8s.io/tiller-user-binding"
[ℹ]  created "flux:Deployment.apps/memcached"
[ℹ]  created "flux:ServiceAccount/flux-helm-operator"
[ℹ]  created "ClusterRole.rbac.authorization.k8s.io/flux-helm-operator"
[ℹ]  created "ClusterRoleBinding.rbac.authorization.k8s.io/flux-helm-operator"
[ℹ]  created "flux:Deployment.apps/flux-helm-operator"
[ℹ]  created "flux:Service/memcached"
[ℹ]  created "flux:ServiceAccount/flux"
[ℹ]  created "ClusterRole.rbac.authorization.k8s.io/flux"
[ℹ]  created "ClusterRoleBinding.rbac.authorization.k8s.io/flux"
[ℹ]  Waiting for Helm Operator to start
ERROR: logging before flag.Parse: E0830 18:22:13.070336   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:13 socat[22982] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:15.095049   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:15 socat[22983] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:17.118096   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:17 socat[22984] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:19.151794   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:19 socat[22985] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:21.181818   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:21 socat[23091] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:23.212878   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:23 socat[23144] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:25.261992   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:25 socat[23158] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:27.282289   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:27 socat[23159] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:29.310963   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:29 socat[23251] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:31.336414   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:31 socat[23377] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:33.361673   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:33 socat[23422] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
ERROR: logging before flag.Parse: E0830 18:22:35.381111   37440 portforward.go:331] an error occurred forwarding 54753 -> 3030: error forwarding port 3030 to pod 88cf78b46bd3d6ae7e949e78440722c086dd7a009397c7a77ef173bf3aac3752, uid : exit status 1: 2019/08/30 09:22:35 socat[23502] E connect(5, AF=2 127.0.0.1:3030, 16): Connection refused
[!]  Helm Operator is not ready yet (Get http://127.0.0.1:54753/healthz: EOF), retrying ...
[ℹ]  Helm Operator started successfully
[ℹ]  see https://docs.fluxcd.io/projects/helm-operator for details on how to use the Helm Operator
[ℹ]  Waiting for Flux to start
[ℹ]  Flux started successfully
[ℹ]  see https://docs.fluxcd.io/projects/flux for details on how to use Flux
[ℹ]  Committing and pushing manifests to git@github.com:marccarre/my-gitops-repo.git
[master eb66e1f] Add Initial Flux configuration
 Author: Flux <>
 13 files changed, 727 insertions(+)
 create mode 100644 flux/flux-account.yaml
 create mode 100644 flux/flux-deployment.yaml
 create mode 100644 flux/flux-helm-operator-account.yaml
 create mode 100644 flux/flux-helm-release-crd.yaml
 create mode 100644 flux/flux-namespace.yaml
 create mode 100644 flux/flux-secret.yaml
 create mode 100644 flux/helm-operator-deployment.yaml
 create mode 100644 flux/memcache-dep.yaml
 create mode 100644 flux/memcache-svc.yaml
 create mode 100644 flux/tiller-ca-cert-configmap.yaml
 create mode 100644 flux/tiller-dep.yaml
 create mode 100644 flux/tiller-rbac.yaml
 create mode 100644 flux/tiller-svc.yaml
Enumerating objects: 17, done.
Counting objects: 100% (17/17), done.
Delta compression using up to 8 threads
Compressing objects: 100% (15/15), done.
Writing objects: 100% (16/16), 8.62 KiB | 8.62 MiB/s, done.
Total 16 (delta 1), reused 11 (delta 1)
remote: Resolving deltas: 100% (1/1), done.
To github.com:marccarre/my-gitops-repo.git
   0bde981..eb66e1f  master -> master
[ℹ]  Flux will only operate properly once it has write-access to the Git repository
[ℹ]  please configure git@github.com:marccarre/my-gitops-repo.git so that the following Flux SSH public key has write access to it
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC1KMlHhlPkDpCLqxvtXexvYE1uHfMjTgdG/mJumlKaHImCq7Z5gagVMolCcTPIMjjqLPRJ0eZnJ+0GWHCpjytr4S1iaiuRddtvOA2WA8nyhcloNXHUevOdeUFvNXFT5LLn9nqYSXU3/yVoP30/LgVsRYPxNi9bH1uEzsoRgIYfKz8qrhRUxfD1Dmp7m5IcwYK1TJeuSvlyGQXVlJQ4o5JJBy/T5eHNhZMsBlX6xFC/P/+fL9PURhyR6hvBJjuhu2BS+2zKLY3yuj9RgYZIMbd43hXEF1ZUOKLh3cmAukjq7ovxDpdp2QCFpk8djTXQHAYfMrq8b1N5OaHoZ/5/iQB7
Cloning into 'my-gitops-repo'...
remote: Enumerating objects: 70, done.
remote: Counting objects: 100% (70/70), done.
remote: Compressing objects: 100% (55/55), done.
remote: Total 212 (delta 16), reused 69 (delta 15), pack-reused 142
Receiving objects: 100% (212/212), 94.32 KiB | 294.00 KiB/s, done.
Resolving deltas: 100% (56/56), done.
[ℹ]  cloning repository "git@github.com:weaveworks/eks-quickstart-app-dev.git":master
Cloning into '/var/folders/24/d3mml6bn20nftpt91cfldq1h0000gn/T/quickstart-512938139'...
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 159 (delta 2), reused 1 (delta 0), pack-reused 150
Receiving objects: 100% (159/159), 39.55 KiB | 257.00 KiB/s, done.
Resolving deltas: 100% (68/68), done.
[ℹ]  processing template files in repository
[ℹ]  writing new manifests to "my-gitops-repo/base"
[master 4ff0501] Add app-dev quickstart components
 Author: Flux <>
 27 files changed, 1241 insertions(+)
 create mode 100644 base/LICENSE
 create mode 100644 base/README.md
 create mode 100644 base/amazon-cloudwatch/cloudwatch-agent-configmap.yaml
 create mode 100644 base/amazon-cloudwatch/cloudwatch-agent-daemonset.yaml
 create mode 100644 base/amazon-cloudwatch/cloudwatch-agent-rbac.yaml
 create mode 100644 base/amazon-cloudwatch/fluentd-configmap-cluster-info.yaml
 create mode 100644 base/amazon-cloudwatch/fluentd-configmap-fluentd-config.yaml
 create mode 100644 base/amazon-cloudwatch/fluentd-daemonset.yaml
 create mode 100644 base/amazon-cloudwatch/fluentd-rbac.yaml
 create mode 100644 base/demo/helm-release.yaml
 create mode 100644 base/kube-system/alb-ingress-controller-deployment.yaml
 create mode 100644 base/kube-system/alb-ingress-controller-rbac.yaml
 create mode 100644 base/kube-system/cluster-autoscaler-deployment.yaml
 create mode 100644 base/kube-system/cluster-autoscaler-rbac.yaml
 create mode 100644 base/kubernetes-dashboard/dashboard-metrics-scraper-deployment.yaml
 create mode 100644 base/kubernetes-dashboard/dashboard-metrics-scraper-service.yaml
 create mode 100644 base/kubernetes-dashboard/kubernetes-dashboard-configmap.yaml
 create mode 100644 base/kubernetes-dashboard/kubernetes-dashboard-deployment.yaml
 create mode 100644 base/kubernetes-dashboard/kubernetes-dashboard-rbac.yaml
 create mode 100644 base/kubernetes-dashboard/kubernetes-dashboard-secrets.yaml
 create mode 100644 base/kubernetes-dashboard/kubernetes-dashboard-service.yaml
 create mode 100644 base/monitoring/metrics-server.yaml
 create mode 100644 base/monitoring/prometheus-operator.yaml
 create mode 100644 base/namespaces/amazon-cloudwatch.yaml
 create mode 100644 base/namespaces/demo.yaml
 create mode 100644 base/namespaces/kubernetes-dashboard.yaml
 create mode 100644 base/namespaces/monitoring.yaml
Enumerating objects: 37, done.
Counting objects: 100% (37/37), done.
Delta compression using up to 8 threads
Compressing objects: 100% (27/27), done.
Writing objects: 100% (36/36), 11.70 KiB | 11.70 MiB/s, done.
Total 36 (delta 8), reused 29 (delta 8)
remote: Resolving deltas: 100% (8/8), done.
To github.com:marccarre/my-gitops-repo.git
   eb66e1f..4ff0501  master -> master
```
Which yielded:
- In Git: https://github.com/marccarre/my-gitops-repo/commit/4ff050153421991c884fe0aa92c71c391e48b004
- In Kubernetes: 
```console
$ kubectl get po --all-namespaces
NAMESPACE              NAME                                                      READY   STATUS    RESTARTS   AGE
amazon-cloudwatch      cloudwatch-agent-h2cs7                                    1/1     Running   0          4m3s
amazon-cloudwatch      fluentd-cloudwatch-dcdzx                                  1/1     Running   0          4m3s
demo                   podinfo-75b8547f78-lvgwx                                  1/1     Running   0          2m19s
flux                   flux-68f59db764-t6wqv                                     1/1     Running   0          5m56s
flux                   flux-helm-operator-6bc7c85bb5-gzbtp                       1/1     Running   0          5m57s
flux                   memcached-958f745c-dcf66                                  1/1     Running   0          5m57s
flux                   tiller-deploy-7ccc4b4d45-9kt5g                            1/1     Running   0          5m56s
kube-system            alb-ingress-controller-78fdc48dd8-dgwtx                   1/1     Running   0          4m3s
kube-system            aws-node-nltbf                                            1/1     Running   0          118m
kube-system            cluster-autoscaler-6c458dd998-gr8z4                       1/1     Running   0          4m3s
kube-system            coredns-5485455cb4-kfwgh                                  1/1     Running   0          124m
kube-system            coredns-5485455cb4-qzb2v                                  1/1     Running   0          124m
kube-system            kube-proxy-9j7j8                                          1/1     Running   0          118m
kubernetes-dashboard   dashboard-metrics-scraper-f7b5dbf7d-l2hqz                 1/1     Running   0          4m3s
kubernetes-dashboard   kubernetes-dashboard-7447f48f55-26dsc                     1/1     Running   0          4m3s
monitoring             alertmanager-prometheus-operator-alertmanager-0           2/2     Running   0          98s
monitoring             metrics-server-7dfc675884-wfksw                           1/1     Running   0          3m59s
monitoring             prometheus-operator-grafana-9bb769cf-52sv7                2/2     Running   0          2m5s
monitoring             prometheus-operator-kube-state-metrics-79f476bff6-9svkn   1/1     Running   0          2m5s
monitoring             prometheus-operator-operator-58fcb66576-4w5k5             1/1     Running   0          2m5s
monitoring             prometheus-operator-prometheus-node-exporter-xdd66        1/1     Running   0          2m5s
monitoring             prometheus-prometheus-operator-prometheus-0               3/3     Running   1          91s
```

Following [Amazon's ALB Ingress docs](https://aws.amazon.com/blogs/opensource/kubernetes-ingress-aws-alb-ingress-controller/):
```console
$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-alb-ingress-controller/v1.0.0/docs/examples/2048/2048-namespace.yaml
$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-alb-ingress-controller/v1.0.0/docs/examples/2048/2048-deployment.yaml
$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-alb-ingress-controller/v1.0.0/docs/examples/2048/2048-service.yaml
$ cat | kubectl apply -f - <<EOF
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: "2048-ingress"
  namespace: "2048-game"
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internet-facing
  labels:
    app: "2048"
spec:
  rules:
    - http:
        paths:
          - path: /*
            backend:
              serviceName: "service-2048"
              servicePort: 80
EOF
ingress.extensions/2048-ingress created

$ kubectl get ingress --all-namespaces
NAMESPACE   NAME              HOSTS   ADDRESS                                                                       PORTS   AGE
2048-game   2048-ingress      *       18967abc-2048game-2048ingr-6fa0-1779707688.ap-northeast-1.elb.amazonaws.com   80      32s
```

And hitting: 18967abc-2048game-2048ingr-6fa0-1779707688.ap-northeast-1.elb.amazonaws.com
yielded:

Screenshot 2019-08-30 at 19 25 31

File: pkg/git/git.go (hunk 4 above)
Comment (marccarre) [reply pkg/git/git.go]: `IsGitURL` is already called in a previous check, before reaching this part.

File: pkg/ctl/install/flux.go (hunk 3 above)
Comment (marccarre) [reply pkg/ctl/install/flux.go]: We have `e.g.:` in other flags' descriptions. I'll harmonise this.

File: pkg/git/git.go (hunk 4 above)
Comment (cPu1) [reply pkg/git/git.go]: I did see that. I was suggesting to somehow use the URL from that call and check its scheme.

Comment (marccarre) [reply pkg/git/git.go]: I'd rather keep the two functions separate for now, but I'm doing this now: https://github.com/weaveworks/eksctl/pull/1232/commits/efbf581aa04517d010a73d3278f419288c141801 to avoid reinventing the wheel and potentially missing edge cases.

Comment (marccarre) [reply pkg/git/git.go]: I'd rather keep the two functions separate for now, but I'm doing this now: https://github.com/weaveworks/eksctl/pull/1232/commits/efbf581aa04517d010a73d3278f419288c141801 to avoid reinventing the wheel and potentially missing edge cases. WDYT?

Comment (marccarre) [reply pkg/git/git.go]: FYI: the SSH URL used in the tests was having a scheme set to `"git"`, hence the current conditionals.
